web/src/main/java/com/es/phoneshop/web/controller/validation/OrderDtoValidator.java

```diff
@@ -0,0 +1,66 @@
+package com.es.phoneshop.web.controller.validation;
+
+import com.es.core.dao.StockDao;
+import com.es.core.model.order.Order;
+import com.es.core.model.order.OrderItem;
+import com.es.core.model.phone.Stock;
+import com.es.phoneshop.web.controller.dto.OrderDto;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+import javax.annotation.Resource;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class OrderDtoValidator implements Validator {
+
+    @Resource
+    private StockDao stockDao;
+
+    @Override
+    public boolean supports(Class<?> aClass) {
+        return Order.class.equals(aClass);
```

Comment:
Bad approach to operate model level classes on view level, hint for future, better to create dedicated dto objects